### ISSUE TITLE
Element disabled when activated seems to prevent aria-live region from announcing, first time only

### ISSUE THREAD

**Issue by brennanyoung** [nvaccess/nvda#9550]
We have a some buttons that make things happen in a canvas. 

Let's just say one of the buttons is "laugh".

The changes in the canvas are announced via aria-live. "Laughter" etc. 

When the laughing is happening - which is quite brief - the "Laugh" button is temporarily disabled. (We don't want to retrigger a laugh if one is already happening. And we don't want the laugh button to "toggle the laugh off"). The laughter should continue without interruption until complete. After that, the button becomes available again.

NVDA announces the "Laugh" button's accessible name when it comes into focus, but when you click/activate it, the first time it says "unavailable" (reflecting the new disabled state, I suppose), but the aria-live region "Laughter" is not announced.

Subsequent activations of the same button announce "unavailable" and then "Laughter". We managed to choreograph the setting of the disabled attribute (using setTimeout) so that it says "unavailable" after "Laughter", but we're wondering why it fails to announce "Laughter" the first time.

In other words, you 'click' a button and NVDA says "unavailable" but -first time only- it doesn't announce "Laughter" which would inform the user that the button behaved as promised.

The aria-live region is present in the static html from the start, and it has been changed at least once before the button gets pressed, because it delivers a kind of "welcome" announcement right at the start.

Is this really how NVDA is supposed to behave?

**Comment by Brian1Gaff** [comment]
Could this be browser specific perhaps?
 I'm no expert but do know that some web sites behave differently with 
different browsers.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by DrSooom** [comment]
@brennanyoung: Please add the NVDA version and all webbrowsers (incl. version), where this behaviour occurs, to the issue description. Thanks.

**Comment by Adriani90** [comment]
We don't have any minimal test case to investigate further, and we didn't hear any updates from the initial author since many years. Closing as abandoned. if you are still having this issue with NvDA 2024.2, please comment and provide a minimal test case. Then we can reopen.
